core/cautils/getter/json_test.go

```diff
@@ -29,4 +29,19 @@ func TestJSONDecoder(t *testing.T) {
 		var receiver string
 		require.Error(t, d.Decode(&receiver))
 	})
+
+	t.Run("empty string input should return EOF error", func(t *testing.T) {
+		const input = ``
+		d := JSONDecoder(input)
+		var receiver string
+		require.Error(t, d.Decode(&receiver))
+		require.Equal(t, "", receiver)
+	})
+
+	t.Run("return error if input string is not a valid json", func(t *testing.T) {
+		const input = `{"name": "John", "age": 30,}`
```

Comment:
maybe you can put `not a valid json` as the string to clearly show it is not valid